src/main/java/dinkplugin/util/ItemSearcher.java

```diff
@@ -0,0 +1,132 @@
+ * <p>
+ * Unlike {@link net.runelite.client.game.ItemManager#search(String)}, this mapping supports untradable items.
+ */
+@Slf4j
+@Singleton
+public class ItemSearcher {
+    private final Map<String, Integer> itemIdByName = Collections.synchronizedMap(new HashMap<>(16384));
+    private @Inject OkHttpClient httpClient;
+    private @Inject Gson gson;
+
+    /**
+     * @param name the exact in-game name of an item
+     * @return the id associated with the item name, or null if not found
+     */
+    @Nullable
+    public Integer findItemId(@NotNull String name) {
+        return itemIdByName.get(name);
+    }
+
+    /**
+     * Begins the initialization process for {@link #itemIdByName}
+     * by querying item names and noted item ids from the RuneLite API,
+     * before passing them to {@link #populate(Map, Set)}
+     *
+     * @implNote This operation does not block the current thread,
+     * by utilizing OkHttp's thread pool and Java's Fork-Join common pool.
+     */
+    @Inject
+    void init() {
+        queryNamesById().thenAcceptBothAsync(
+            queryNotedItemIds().exceptionally(throwable -> Collections.emptySet()),
+            this::populate
+        );
+    }
+
+    /**
+     * Populates {@link #itemIdByName} with the inverted mappings of {@code namesById},
+     * while skipping noted items specified in {@code notedIds}.
+     *
+     * @param namesById a mapping of item id's to the corresponding in-game name
+     * @param notedIds  the id's of noted items
+     * @implNote When multiple non-noted item id's have the same in-game name, only the earliest id is saved
+     */
+    @VisibleForTesting
+    void populate(@NotNull Map<Integer, String> namesById, @NotNull Set<Integer> notedIds) {
+        namesById.forEach((id, name) -> {
+            if (!notedIds.contains(id))
+                itemIdByName.putIfAbsent(name, id);
+        });
+
+        log.debug("Completed initialization of item cache with {} entries", itemIdByName.size());
+    }
+
+    /**
+     * @return a mapping of item ids to their in-game names, provided by the RuneLite API
+     */
+    private CompletableFuture<Map<Integer, String>> queryNamesById() {
+        return queryCache("names.json", new TypeToken<Map<Integer, String>>() {});
+    }
+
+    /**
+     * @return a set of id's of noted items, provided by the RuneLite API
+     */
+    private CompletableFuture<Set<Integer>> queryNotedItemIds() {
+        return queryCache("notes.json", new TypeToken<Map<Integer, Integer>>() {})
+            .thenApply(Map::keySet);
+    }
+
+    /**
+     * @param fileName the name of the file to query from RuneLite's cache
+     * @param type     a type token that indicates how the json response should be parsed
+     * @return the transformed cache response, wrapped in a future
+     */
+    private <T> CompletableFuture<T> queryCache(@NotNull String fileName, @NotNull TypeToken<T> type) {
```

Comment:
Should this be moved to `ItemUtils` as: `readJson(httpClient, gson, url, type)` ?